### ISSUE TITLE
C2C needs to take arguments

### ISSUE THREAD

**Issue by grav3m1nd-byte** [SaumyajeetDas/GodGenesis#2]
At the moment, I started testing this out after reading about it and quite frankly looks promising.

After a brief review, one thing that C2C should have is that at least it takes arguments so users provides IP address and/interface to bind to plus the port.

At the moment, my time is slightly limited, but I can contribute and take care of that part in like two weeks.

**Comment by SaumyajeetDas** [comment]
Sure.

**Comment by ajarham14** [comment]
Alright but how I run it in termux,

Guide me plz

**Comment by ajarham14** [comment]
> Sure.

How I run it in termux app.. 
Plz tell me. 
 It may be working in termux. Or not

**Comment by rly0nheart** [comment]
Hello, can I modify it and add arguments as requested in issue #2 ?, I will open a pull request once I finish

**Comment by SaumyajeetDas** [comment]
> Hello, can I modify it and add arguments as requested in issue #2 ?, I will open a pull request once I finish

Sure.

**Comment by grav3m1nd-byte** [comment]
> Hello, can I modify it and add arguments as requested in issue #2 ?, I will open a pull request once I finish

Hi @rly0nheart, if you can take care of it, I'd say go for it. I don't have the bandwidth right now to work on this so feel free to jump ahead.

**Comment by grav3m1nd-byte** [comment]
> > Sure.
> 
> How I run it in termux app.. Plz tell me. It may be working in termux. Or not

Hi there,

I am not sure if it running on termux or not should be part of this Issue where we are looking to have the C2C script accept sysargs instead of having to modify the script. Not too familiar with termux, but should run similarly to any other terminar emulator.

**Comment by grav3m1nd-byte** [comment]
@SaumyajeetDas and @rly0nheart, I can open a new issue for this that i'm gonna say, but it would be nice if the payload script could be used remotely as well and/or to simply create a payload that doesn't rely on loading the script on the target machine and then installing python. What do you guys think?

**Comment by rly0nheart** [comment]
> > Hello, can I modify it and add arguments as requested in issue #2 ?, I will open a pull request once I finish
> 
> Hi @rly0nheart, if you can take care of it, I'd say go for it. I don't have the bandwidth right now to work on this so feel free to jump ahead.

I made the changes, and added a few improvements, I thought it would be great if the payload got generated by passing the right argument, so I added a payload_builder.py file.

## Note
> I was making the changes on windows and the c2c.py and the payload.py got detected by my Bitdefender

**Comment by grav3m1nd-byte** [comment]
> > > Hello, can I modify it and add arguments as requested in issue #2 ?, I will open a pull request once I finish
> > 
> > 
> > Hi @rly0nheart, if you can take care of it, I'd say go for it. I don't have the bandwidth right now to work on this so feel free to jump ahead.
> 
> I made the changes, and added a few improvements, I thought it would be great if the payload got generated by passing the right argument, so I added a payload_builder.py file.
> 
> ## Note
> > I was making the changes on windows and the c2c.py and the payload.py got detected by my Bitdefender

Yeah, I realized you already worked on this. I'm reviewing these changes and I like it.

If these got detected, it might be a matter of time before other AVs start detecting it...maybe not.

**Comment by rly0nheart** [comment]
> > > Sure.
> > 
> > 
> > How I run it in termux app.. Plz tell me. It may be working in termux. Or not
> 
> Hi there,
> 
> I am not sure if it running on termux or not should be part of this Issue where we are looking to have the C2C script accept sysargs instead of having to modify the script. Not too familiar with termux, but should run similarly to any other terminar emulator.


> @SaumyajeetDas and @rly0nheart, I can open a new issue for this that i'm gonna say, but it would be nice if the payload script could be used remotely as well and/or to simply create a payload that doesn't rely on loading the script on the target machine and then installing python. What do you guys think?

If the payload got compiled (with pyinstaller or an alternative), it would run without requiring python on the target

**Comment by rly0nheart** [comment]
> > > > Hello, can I modify it and add arguments as requested in issue #2 ?, I will open a pull request once I finish
> > > 
> > > 
> > > Hi @rly0nheart, if you can take care of it, I'd say go for it. I don't have the bandwidth right now to work on this so feel free to jump ahead.
> > 
> > 
> > I made the changes, and added a few improvements, I thought it would be great if the payload got generated by passing the right argument, so I added a payload_builder.py file.
> > ## Note
> > > I was making the changes on windows and the c2c.py and the payload.py got detected by my Bitdefender
> 
> Yeah, I realized you already worked on this. I'm reviewing these changes and I like it.
> 
> If these got detected, it might be a matter of time before other AVs start detecting it...maybe not.

That's a great idea, I did not get to test the changes properly, so there might be some bugs

**Comment by rly0nheart** [comment]
Let's hope they don't get detected by other AV's👀😅

**Comment by grav3m1nd-byte** [comment]
> > > > > Hello, can I modify it and add arguments as requested in issue #2 ?, I will open a pull request once I finish
> > > > 
> > > > 
> > > > Hi @rly0nheart, if you can take care of it, I'd say go for it. I don't have the bandwidth right now to work on this so feel free to jump ahead.
> > > 
> > > 
> > > I made the changes, and added a few improvements, I thought it would be great if the payload got generated by passing the right argument, so I added a payload_builder.py file.
> > > ## Note
> > > > I was making the changes on windows and the c2c.py and the payload.py got detected by my Bitdefender
> > 
> > 
> > Yeah, I realized you already worked on this. I'm reviewing these changes and I like it.
> > If these got detected, it might be a matter of time before other AVs start detecting it...maybe not.
> 
> That's a great idea, I did not get to test the changes properly, so there might be some bugs

Where did you make the changes? on VIM or VSCode? If you used VIM, you can use pylint and at least use a linter (it won't necessarily help with bugs but it's something).

**Comment by grav3m1nd-byte** [comment]
> Let's hope they don't get detected by other AV'seyessweat_smile

Honestly, it is just room for improvement, or at least that what it means to me. The fact that this project was created and works is cool enough.

You know, it is possible it got flagged because fodhelper.exe exists in the code, and that is now known to be used for UAC bypass. If so, that is another reason to get this project to create custom payloads offline, instead of loading the script into the victim machine to then use it.

BTW, try and test [Hoaxshell](https://github.com/t3l3machus/hoaxshell). It does the try and pretty good.

**Comment by SaumyajeetDas** [comment]
Give me a bit guys.
Me and @ANON4MOUS is thinking to collaborate with you. Can we create one group we just want to discuss few things.

Please do share your thoughts guys.
Looking forward to work with you.

**Comment by rly0nheart** [comment]
> > Let's hope they don't get detected by other AV'seyessweat_smile
> 
> Honestly, it is just room for improvement, or at least that what it means to me. The fact that this project was created and works is cool enough.
> 
> You know, it is possible it got flagged because fodhelper.exe exists in the code, and that is now known to be used for UAC bypass. If so, that is another reason to get this project to create custom payloads offline, instead of loading the script into the victim machine to then use it.

Agreed!, Also I made the changes on VS Code, I did not encounter any syntax errors, but you never know

**Comment by grav3m1nd-byte** [comment]
> Give me a bit guys. Me and @ANON4MOUS is thinking to collaborate with you. Can we create one group we just want to discuss few things.

Sure thing...I mean, I just got wrapped up in thinking about the potential here :smile:

**Comment by SaumyajeetDas** [comment]
Thanks for your help.
Please do tell your preferable medium for the group.

**Comment by grav3m1nd-byte** [comment]
> Thanks for your help. Please do tell your preferable medium for the group.

I'm open to anything...Discord maybe?

**Comment by SaumyajeetDas** [comment]
No problem at our end.
Lets wait for @rly0nheart

**Comment by rly0nheart** [comment]
Discord is good 🔥

**Comment by rly0nheart** [comment]
> No problem at our end. Lets wait for @rly0nheart

Sure, create the group and add me, rly0nheart#2779

**Comment by rly0nheart** [comment]
> > Let's hope they don't get detected by other AV'seyessweat_smile
> 
> Honestly, it is just room for improvement, or at least that what it means to me. The fact that this project was created and works is cool enough.
> 
> You know, it is possible it got flagged because fodhelper.exe exists in the code, and that is now known to be used for UAC bypass. If so, that is another reason to get this project to create custom payloads offline, instead of loading the script into the victim machine to then use it.
> 
> BTW, try and test [Hoaxshell](https://github.com/t3l3machus/hoaxshell). It does the try and pretty good.

Just looked at hoaxshell, looks good!

**Comment by rly0nheart** [comment]
> > rly0nheart#2779
> 
> @grav3m1nd-byte @rly0nheart Guys can u please join from the following link. https://discord.gg/HcJUTHz

The link says its invalid

**Comment by SaumyajeetDas** [comment]
> > > rly0nheart#2779
> > 
> > @grav3m1nd-byte @rly0nheart Guys can u please join from the following link. https://discord.gg/HcJUTHz
> 
> The link says its invalid

https://discord.gg/rrzhjr33

Here is the new link.
